src/client/game.cpp

```diff
@@ -1384,7 +1384,15 @@ bool Game::createSingleplayerServer(const std::string &map_dir,
 {
 	showOverlayMessage(N_("Creating server..."), 0, 5);
 
-	std::string bind_str = g_settings->get("bind_address");
+	std::string bind_str;
+	if (simple_singleplayer_mode) {
+		// Make the simple singleplayer server only accept connections from localhost,
+		// which also makes Windows Defender not show a warning.
+		bind_str = "127.0.0.1";
```

Comment:
could be "localhost" to leverage ipv6 if that's enabled

Reply:
That does not work for me (Windows 10).

-------

I can ping `::1` though, so this is not a Windows configuration problem.

<details>
<summary>Pinging ::1</summary>

```
C:\>ping ::1

Pinging ::1 with 32 bytes of data:
Reply from ::1: time<1ms
Reply from ::1: time<1ms
Reply from ::1: time<1ms
Reply from ::1: time<1ms

Ping statistics for ::1:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 0ms, Maximum = 0ms, Average = 0ms
```
</details>

<details>
<summary>Pinging localhost</summary>

```
C:\>ping localhost

Pinging LAPTOP [::1] with 32 bytes of data:
Reply from ::1: time<1ms
Reply from ::1: time<1ms
Reply from ::1: time<1ms
Reply from ::1: time<1ms

Ping statistics for ::1:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 0ms, Maximum = 0ms, Average = 0ms
```
</details>

Reply:
That does not work for me (Windows 10).

-------

I can `ping ::1` though. Pinging `localhost` gives me `::1` address, too.

Reply:
Looks like the client code always connects via IPv4 so that indeed might not work: https://github.com/minetest/minetest/blob/03ba9370b951f1c0904b53dfac3f5daf0bb354d6/src/client/game.cpp#L1554-L1568

Not important anyway.

Reply:
Yeah if it's set to `localhost` and it resolves to IPv6, and ipv6_server isn't enabled it will just freeze. I'm unsure if there's any performance difference between connecting with IPv4 or IPv6 over localhost (probably not), but always connecting to singleplayer over IPv4 is probably good for now.